### DISCUSSION TITLE
Add "download all" button in playlist rooms

### DISCUSSION THREAD

**Discussion by Gonzalo-Bruna** [ppy/osu#13204] [Ideas] [answered]
**Describe the new feature:** 

I searched for opened issues but I couldn't find it.

It would be very helpful to add a "download all" button in playlist rooms, since some of them have lots of beatmaps and downloading them all at once would be a good feature to have.

**Proposal designs of the feature:** 

Maybe a button here?

![Inkedosu_2021-05-18_15-37-06_LI](https://user-images.githubusercontent.com/41755816/118713253-b6fc8000-b7ef-11eb-9738-372b8629195c.jpg)

**Comment by Loreos7** [comment]


**Reply by frenzibyte** [reply]
Please refrain from such comments as it helps nothing but noise. Upvote the discussion if you do want to raise it up.

**Comment by peppy** [comment, marked as answer]
Considering that playlists can contain hundreds of beatmaps currently, this can't be implemented as things stand. Maybe we can have auto-download on selecting a playlist item?

**Reply by Moltanicaa** [reply]
Can you make it possible to select maps using shift?

**Reply by peppy** [reply]
You could just click down the list in that case?

**Reply by Moltanicaa** [reply]
Yes

**Reply by Loreos7** [reply]
If I want to play the entire playlist, then there must be a button that downloads all the maps. It seems to me that you are trying to implement a feature that downloads all the maps at once. This is a really bad idea for the large playlists. But why download everything at once when you can download one at a time? 
So i click just one button and download starts one-by-one instead of clicking >20 buttons

**Reply by peppy** [reply]
As proposed, if this is implemented it would be automatic download after clicking the item. Not the whole playlist.

No further discussion is required.

**Reply by TheFantasticLoki** [reply]
I feel also adding a button to download the first say 10 missing maps on each click to minimise server load on massive playlists while still being able to download multiple maps at a time.

**Reply by n0Oo0Oo0b** [reply]
I agree with Loreos7 here: what's wrong with a buffered 'download all' button which is limited to 3-5 downloads at a time when you can easily get 10+ concurrent downloads by manually clicking each button?
